### PR TITLE
Initial work to get pip working on Mono

### PR DESCRIPTION
**Pull request by slide**

### CODE REVIEW

**Comment by slozier** [comment]
There's also `PipeStream` in `PythonFile.TryGetFileHandle`. Since they're no longer PipeStreams you may need to add a check.

File: .gitignore [hunk 1]
```diff
@@ -48,6 +49,9 @@ Tests/compat/tempfile.txt
 Tests/Tools/stdmodules.py.stdout.log
 Tests/test_data.gz
 
+# nunit
+Util/References/NUnit*/
```
Comment (slozier) [inline .gitignore]: I thought this was already in .gitignore?

Comment (slide) [inline .gitignore]: It was showing up for me when I did a git status

Comment (slozier) [inline .gitignore]: Looks like your branch is old. Maybe try rebasing?

Comment (slide) [inline .gitignore]: I pulled that morning, but can double check

**Pull request merged**